### PR TITLE
feat: confirm seed phrase with multiple-choice

### PR DESCRIPTION
**Pull request by Fbin243**

### CODE REVIEW

File: src/icons/ArrowRight.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { FC } from "react";
+import { IconProps } from "./types";
+
+const ArrowRight: FC<IconProps> = ({ size = 20, color = "#267578" }) => {
```
Comment (devtronghieu) [inline src/icons/ArrowRight.tsx:4]: rename to `ArrowRightIcon`

File: src/icons/Correct.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { FC } from "react";
+import { IconProps } from "./types";
+
+const Correct: FC<IconProps> = ({ size = 20, color = "#267578" }) => {
```
Comment (devtronghieu) [inline src/icons/Correct.tsx:4]: same here, `CorrectIcon` please

File: src/icons/Correct.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  return (
+    <svg width={size} height={size} viewBox="0 0 14 9" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (devtronghieu) [inline src/icons/Correct.tsx:6]: as the `viewBox` has the `width` is 14 and `height` is `9`, you should change the height's value to `size * 9 / 14`

File: src/icons/ShieldFail.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { FC } from "react";
+import { IconProps } from "./types";
+
+const ArrowRight: FC<IconProps> = ({ size = 20, color = "#267578" }) => {
```
Comment (devtronghieu) [inline src/icons/ShieldFail.tsx:4]: `ShieldFailIcon` instead of `ArrowRight`

File: src/icons/ShieldFail.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+    <svg width={size} height={size} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+      <g id="Iconly/Bold/Shield Fail">
+        <g id="Shield Fail">
+          <path
+            id="Shield Fail_2"
+            fill-rule="evenodd"
+            clip-rule="evenodd"
```
Comment (devtronghieu) [inline src/icons/ShieldFail.tsx:12]: change `-rule` to `Rule`

File: src/icons/Wrong.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { FC } from "react";
+import { IconProps } from "./types";
+
+const Wrong: FC<IconProps> = ({ size = 20, color = "#EC044E" }) => {
```
Comment (devtronghieu) [inline src/icons/Wrong.tsx:4]: `WrongIcon`

File: src/screens/CreateWallet/Dialog.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      <div className="w-60 rounded-lg overflow-hidden shadow-xl">
+        <div className="flex items-center  bg-primary-300 text-secondary-200 px-4 py-2">
+          <ShieldFail size={24} color="#FFDFBE" />
+          <p className=" font-semibold ms-1">Warning!!</p>
+        </div>
+        <div className=" bg-primary-100 p-4">
+          <p className="dialog-description  text-primary-500 font-medium">
+            We suggest you to view your seed phrase again and store it carefully before going through these tests.
+          </p>
+          <div className=" flex justify-between mt-4">
```
Comment (devtronghieu) [inline src/screens/CreateWallet/Dialog.tsx:21]: there are some `className` start with a space ` `, I think we remove it

File: src/screens/CreateWallet/ConfirmSeedPhrase.tsx [hunk 8]
```diff
@@ -1,59 +1,146 @@
...
+        <div className="flex justify-between">
+          <button
+            className={`rounded-button ${currentQuestionIndex === 0 ? "bg-gray-200" : "bg-primary-200"}`}
+            onClick={!isChosen || isCorrect ? handlePrevQuestion : () => {}}
+          >
+            <ArrowLeft size={18} />
+          </button>
+          <button
+            className={`rounded-button ${currentQuestionIndex === 2 ? "bg-gray-200" : "bg-primary-200"}`}
+            onClick={!isChosen || isCorrect ? handleNextQuestion : () => {}}
```
Comment (devtronghieu) [inline src/screens/CreateWallet/ConfirmSeedPhrase.tsx:125]: can I change the ternary operator to this `(!isChosen || isCorrect) && handleNextQuestion`

File: src/screens/CreateWallet/ConfirmSeedPhrase.tsx [hunk 9]
```diff
@@ -1,59 +1,146 @@
...
-  const handleInputChange = (index: number) => (e: ChangeEvent<HTMLInputElement>) => {
-    const newWords = new Map(words);
-    newWords.set(index, e.target.value.toLowerCase());
-    setWords(newWords);
-  };
+const ConfirmSeedPhrase: FC<Props> = ({ seedPhraseWords, setIsSeedPhraseConfirmed, onGoBack }) => {
+  const [questions, setQuestions] = useState<Question[]>(() => {
+    const questions: Question[] = [];
+    for (let i = 0; i < 3; i++) {
+      questions.push(newQuestion());
```
Comment (devtronghieu) [inline src/screens/CreateWallet/ConfirmSeedPhrase.tsx:26]: Based on the implementation of the `newQuestion`, I think we have an edge case when an index randomly appears twice. May we have an `existedIndexes` list to re-verify?

Comment (Fbin243) [reply src/screens/CreateWallet/ConfirmSeedPhrase.tsx:26]: Yeah, you're right :> I have't checked it yet

File: src/icons/Wrong.tsx (hunk 6 above)
Comment (Fbin243) [reply src/icons/Wrong.tsx:4]: Do I need to change the file's name ? 
Thanks to the `export default,` I also import `WrongIcon` with the name `Wrong`. Do you want to fix all of them?

Comment (devtronghieu) [reply src/icons/Wrong.tsx:4]: It is just my personal convention, those files are already in the `icons` folder, so we don't have to name the file `NameIcon` one more time, but the name inside I'd prefer it has a `Icon` suffix for easier import with VSCode intellisense. You may use the `F2` key to rename it for all the implemented code.

File: src/screens/CreateWallet/ConfirmSeedPhrase.tsx (hunk 8 above)
Comment (devtronghieu) [reply src/screens/CreateWallet/ConfirmSeedPhrase.tsx:125]: Although it is runnable but yeah, it is considered a bad pattern anyway, just forget my first reply. How about bringing the condition into the function?

```ts
function handlePrevQuestion(): void {
  if ((!isChosen || isCorrect) && currentQuestionIndex > 0) {
    setCurrentQuestionIndex(currentQuestionIndex - 1);
  }
}
```

File: src/icons/Wrong.tsx (hunk 6 above)
Comment (Fbin243) [reply src/icons/Wrong.tsx:4]: yeah, good tip

File: src/screens/CreateWallet/ConfirmSeedPhrase.tsx (hunk 8 above)
Comment (Fbin243) [reply src/screens/CreateWallet/ConfirmSeedPhrase.tsx:125]: good way, it looks like more reasonable

**Review by devtronghieu — APPROVED** [review]

**Pull request merged**